### PR TITLE
Add nitpicky, reformat docs filepath and spell check

### PR DESCRIPTION
**Pull request by nabobalis**
images

TODO:
- [x] Fix the last few issues with nitpicky
- [x] Check through the rendered API pages
- [x] Check the `__all__` for each python file
- [x] Go over the docs again for kicks
- [x] Check that specutils is ok with these changes (ref https://github.com/sunpy/ndcube/issues/604)
- [x] Wonder why I did this and how I will atone for it when I go to Hell.

### CODE REVIEW

File: ndcube/extra_coords/__init__.py
Comment (nabobalis) [inline ndcube/extra_coords/__init__.py:6]: I added to the namespace here.

File: ndcube/ndcollection.py
Comment (nabobalis) [inline ndcube/ndcollection.py:79]: So I for the life of me could not get this attribute to render in sphinx without doing this.

Now, this has to be wrong, there has to be a way to get the `self.whatever` to render in the docs and I don't know what that is.

File: ndcube/utils/__init__.py
Comment (nabobalis) [inline ndcube/utils/__init__.py:1]: Can I ask why this was done?

File: .gitignore
Comment (nabobalis) [inline .gitignore:211]: Revert this

Comment (nabobalis) [inline .gitignore:211]:
```suggestion
docs/generated
docs/api
```

File: ndcube/global_coords.py
Comment (nabobalis) [inline ndcube/global_coords.py:14]: I have done this in a few places and well, my idea is that since this is actually in the main namespace, we want to avoid this all and it causing problems with multiple places a user might import it from.

File: ndcube/ndcube.py
Comment (nabobalis) [inline ndcube/ndcube.py:33]: Same here as in other alls, now these are only normally discoverable from the main ndcube import.

File: ndcube/ndcube_sequence.py
Comment (nabobalis) [inline ndcube/ndcube_sequence.py:11]: Same here as in other alls, now these are only normally discoverable from the main ndcube import.

**Comment by DanRyanIrish** [comment]
@nabobalis this is a monster PR.  Looks like you have a couple other checklist items.  Let me know when you're ready for a review.

**Comment by nabobalis** [comment]
> @nabobalis this is a monster PR. Looks like you have a couple other checklist items. Let me know when you're ready for a review.

This is ready to go. The last checklist item I will need to think about more but the rest is ok.

File: ndcube/ndcollection.py
Comment (DanRyanIrish) [reply ndcube/ndcollection.py:79]: I wonder if changing this to a property should be classes as an API change.  The tests seem to be passing do presumably this doesn't cause internal errors.  But if someone is altering `self.aligned_axes in place, this will cause their workflow to break.  That said, we probably don't want anyone changing those values in place.

Perhaps a breaking changelog entry should be added to this PR saying `self.aligned_axes` is now a property.

File: setup.cfg
Comment (DanRyanIrish) [inline setup.cfg:64]: Is this issue in the comment now resolved?

File: ndcube/visualization/mpl_sequence_plotter.py
Comment (DanRyanIrish) [inline ndcube/visualization/mpl_sequence_plotter.py:116]: What does this change do exactly?

File: ndcube/visualization/mpl_sequence_plotter.py
Comment (DanRyanIrish) [inline ndcube/visualization/mpl_sequence_plotter.py:57]: A bugfix changelog should be added for this change.

File: ndcube/utils/__init__.py
Comment (DanRyanIrish) [reply ndcube/utils/__init__.py:1]: I actually don't know...  I'm happy for it to be changed if it doesn't break any tests.

File: ndcube/ndcube.py
Comment (DanRyanIrish) [inline ndcube/ndcube.py:25]: Why not just `from ndcube import ExtraCoords, ExtraCoordsABC`?

File: ndcube/ndcube.py
Comment (DanRyanIrish) [inline ndcube/ndcube.py:701]:
```suggestion
            A new resultant NDCube object, the supplied ``target_wcs`` will be the ``.wcs`` attribute of the output `~ndcube.NDCube`.
```

File: ndcube/ndcube.py
Comment (DanRyanIrish) [inline ndcube/ndcube.py:959]:
```suggestion
        : `~ndcube.NDCube`
```

File: docs/explaining_ndcube/tabular_coordinates.rst
Comment (DanRyanIrish) [inline docs/explaining_ndcube/tabular_coordinates.rst:67]:
```suggestion
This `gwcs.wcs.WCS` object can then be passed to the constructor of `~ndcube.NDCube` alongside your array and other parameters.
```

File: docs/explaining_ndcube/visualization.rst
Comment (DanRyanIrish) [inline docs/explaining_ndcube/visualization.rst:287]:
```suggestion
To make a 4D array out of the data arrays within the constituent `~ndcube.NDCube` instances in
``my_sequence``.
```

File: docs/extending_ndcube.rst
Comment (DanRyanIrish) [inline docs/extending_ndcube.rst:32]:
```suggestion
Your subclass of `~ndcube.NDCube` may provide great functionality for the users who have data which meet your assumptions.
```

File: docs/extending_ndcube.rst
Comment (DanRyanIrish) [inline docs/extending_ndcube.rst:47]:
```suggestion
On `~ndcube.NDCube` the only method which returns another instance of your subclass is currently `ndcube.NDCube.reproject_to`.
```

File: docs/index.rst
Comment (DanRyanIrish) [inline docs/index.rst:5]:
```suggestion
`ndcube` is a SunPy Project affiliated package designed for handling N-dimensional data cubes described by WCS (World Coordinate System) transformations.
```

File: docs/index.rst
Comment (DanRyanIrish) [inline docs/index.rst:61]:
```suggestion
The only thing you will need to change is "sunpy" to "ndcube" e.g.,
```

**Review by DanRyanIrish — CHANGES_REQUESTED** [review]
Hi @nabobalis.  A mammoth PR.  I think it looks good.  I have only a few (relative to the PR size) comments.  If some of the suggested changes break the doc build then feel free to not use them.

**Comment by DanRyanIrish** [comment]
> This is ready to go. The last checklist item I will need to think about more but the rest is done. One item is asking if you would like a changelog entry.

In the inline comments I have suggested a couple specific changelog entries.

**Comment by DanRyanIrish** [comment]
@nabobalis I just merged PR #609 which made changes to `joss_paper.md` which is also altered by this PR.  So you need to do update this branch from main before merging this PR.

File: ndcube/ndcollection.py
Comment (nabobalis) [reply ndcube/ndcollection.py:79]: I can add a setter and that should avoid a breaking change?

**Comment by nabobalis** [comment]
> Hi @nabobalis. A mammoth PR. I think it looks good. I have only a few (relative to the PR size) comments. If some of the suggested changes break the doc build then feel free to not use them.

Thanks, I will accept them all and work to fix what is broken. I will also rebase ontop of main.

File: ndcube/visualization/mpl_sequence_plotter.py
Comment (nabobalis) [reply ndcube/visualization/mpl_sequence_plotter.py:116]: It just highlights the fact that the last two outputs are unused.

File: ndcube/ndcube.py
Comment (nabobalis) [reply ndcube/ndcube.py:25]: I personally subscribe to the idea that internal code should not call the public API. 

BUt I can make this change.

**Comment by nabobalis** [comment]
About the change to `aligned_axes`, is it meant to be changeable after the fact?

File: ndcube/utils/__init__.py
Comment (nabobalis) [reply ndcube/utils/__init__.py:1]: It is so someone cna import ndcube.utils and get access to the entire range of modules without having to import them directly. This will stay.

**Comment by DanRyanIrish** [comment]
> About the change to `aligned_axes`, is it meant to be changeable after the fact?
> 
> I did add a setter just in case.

At this time, no I don't think it should be changeable after the fact.  Supporting that needs a bit more thought.

**Comment by nabobalis** [comment]
> > About the change to `aligned_axes`, is it meant to be changeable after the fact?
> > I did add a setter just in case.
> 
> At this time, no I don't think it should be changeable after the fact. Supporting that needs a bit more thought and should probably have it's own PR

I will remove this then.

**Comment by nabobalis** [comment]
Now it has been removed.

File: changelog/607.breaking.rst
Comment (nabobalis) [inline changelog/607.breaking.rst:1]:
```suggestion
`ndcube.NDCollection.aligned_axes` attribute is now property.
```

**Comment by DanRyanIrish** [comment]
Let me know if and when you need another review of this.

**Comment by nabobalis** [comment]
> Let me know if and when you need another review of this.

Its ready.

**Comment by DanRyanIrish** [comment]
Hi @nabobalis.  Do you need another review of this or are you confident all issues have been addressed?

**Comment by nabobalis** [comment]
> Hi @nabobalis. Do you need another review of this or are you confident all issues have been addressed?

I think everything is addressed and now ready for another review.

**Comment by nabobalis** [comment]
Thanks for the review and merge @DanRyanIrish!

I have manually backported this and will check the CI is ok on 2.1 and do a release.

**Pull request merged**
